design_docs/ShakaDistanceMatrix.md

```diff
@@ -0,0 +1,55 @@
+# Distance Matrix
+The Distance Matrix is an efficient data structure for fast lookups of edge weights and potential parallelization of edge recomputations. Efficient removal of locations for the application's use case can be made through an additional Hash Table.
+
+## Interface
+
+### protocol ShakaDistanceMatrix
+
+**Methods:**
+- public addLocations(locations: Array[ShakaNode])
+  - **Parameters:**
+    - locations: Array[ShakaNode] -> New location(s) to be added
+  - **Description:**
+    - A list of locations will be added to the ShakaDistanceMatrix with their appropriate edge weights added for all entries as well
+- public getEdgeWeight(source: ShakaNode, dest: ShakaNode) => Float
+  - **Parameters:**
+    - source: ShakaNode -> Source node
+    - dest: ShakaNode -> Destination node
+  - **Returns:**
+    - Edge weight from source to destination
+  - **Description:**
+    - Get a single distance (edge weight) from source to destination.
+- public removeLocations(locations: Array[ShakaNode])
+  - **Parameters:**
+    - locations: Array[ShakaNode] -> Location(s) to be removed
+  - **Description:**
+    - Remove locations from the ShakaDistanceMatrix
+- public updateEdgeWeights(locations: Array[ShakaNode])
+  - **Parameters:**
+    - locations: Array[ShakaNode] -> Location(s) to have their related edge weight(s) updated.
+

```

Comment:
We should have a simple method that checks if the list of locations is empty meaning all inputted locations have been removed.

Reply:
I agree. Added as 
- public isEmpty() => Boolean
  - **Returns:**
    - True if there are no more locations to visit.
  - **Description:**
    - Returns whether or not there are locations that still need to be visited. Implementation will need to somehow keep track of this.

Reply:
Looks good!